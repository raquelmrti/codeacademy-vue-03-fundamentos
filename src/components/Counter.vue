<template>
<h2>Counter</h2>
<p>10</p>

</template>

<script>
export default {
  // name: "Patito"
}
</script>

<style>

</style>