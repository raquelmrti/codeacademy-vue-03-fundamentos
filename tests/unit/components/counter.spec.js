import { shallowMount } from "@vue/test-utils"
import Counter from "@/components/Counter"

describe("Counter Component", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Counter)
  })

  // test("Must match with snapshot", () => {
  //   const wrapper = shallowMount(Counter)
  //   expect(wrapper.html()).toMatchSnapshot()
  // })

  test("h2 should have default value 'Counter'", () => {
    //find will find the first h2, if I had multiple h2 elements
    //I could use findAll
    expect(wrapper.find("h2").exists()).toBeTruthy()

    const h2Value = wrapper.find("h2").text()
    expect(h2Value).toBe("Counter!!")
  })

  test("default value should be 100 in p", () => {
    const value = wrapper.find("[data-testid='counter']").text()
    // console.log(pTags[1].text())
    expect(value).toBe("100")
  })

  test("counter value should increase and decrease by 1", async () => {
    const [increaseBtn, decreaseBtn] = wrapper.findAll("button")

    await increaseBtn.trigger("click")
    await increaseBtn.trigger("click")
    await increaseBtn.trigger("click")

    await decreaseBtn.trigger("click")
    await decreaseBtn.trigger("click")

    const value = wrapper.find("[data-testid='counter']").text()

    expect(value).toBe("101")
  })

  test("should set default value", () => {
    const { start } = wrapper.props()
    const value = wrapper.find("[data-testid='counter']").text()

    expect(Number(value)).toBe(start)
  })

  test("should show the title prop", () => {
    const title = "Hola Mundo"
    const wrapper = shallowMount(Counter, {
      props: {
        title,
        // start: "5"
      },
    })

    const value = wrapper.find("h2").text()

    expect(value).toBe("Hola Mundo!!")
  })
})
