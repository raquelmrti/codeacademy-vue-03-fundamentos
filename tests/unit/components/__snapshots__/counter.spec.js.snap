// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Must match with snapshot 1`] = `
<h2>Counter!!</h2>
<p>100 <sup>6</sup> = 10000</p>
<div><button>+1</button><button>-1</button></div>
`;
